xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-ui/src/main/resources/AnnotationCode/Script.xml

```diff
@@ -1677,6 +1677,12 @@ XWiki.Annotation = Class.create({
           onSuccess: function(response) {
             // display the annotation creation form
             this.fillCreateForm(this.createPanel, response.responseText);
+            // If the WYSIWYG editor used is CKEditor, we make sure its input gets focused as soon as it's loaded
+            // Note that this CKE config is easier than having to handle custom eventListeners.
+            // As of 17.0.0, this does not interfere with the inline editor, so this solution is okay.
+            if(typeof CKEDITOR !== 'undefined') {
```

Comment:
@mflorea I need your input there, is that the proper way to detect CKEditor?

Reply:
Note that this is the same as the way we used at https://github.com/xwiki/xwiki-platform/commit/3b2b6f39ae799438f316a59717e49c04d763e871 . If this is not a correct way, we would need to fix that place too.

According to tests on my local instance this works.

Reply:
Hmm actually in the link you provided @mflorea made some comments about this check and about startupFocus. 
Feels like it could be better to put that in an internal library that we would reuse where needed, and so if we need or want to change the way to perform this, we wouldn't need to check everywhere. 
wdyt? does it worth it?

Reply:
As I said on my comment for https://github.com/xwiki/xwiki-platform/commit/3b2b6f39ae799438f316a59717e49c04d763e871 , I think it's wrong to hard-code the CKEditor usage. The solution should be generic, for any WYSIWYG editor:

* either by setting some attribute on the text area that is enhanced by CKEditor, before the WYSIWYG editor is loaded, and having some code in a CKEditor plugin that reacts to that attribute
* or by triggering some generic event that a CKEditor plugin can listen to and react
* or, if the previous two options are not possible, introduce some generic WYSIWYG editor JavaScript API that we implement for now in a CKEditor plugin, but this requires more work